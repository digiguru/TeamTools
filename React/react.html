<html>
    <head>

           <link rel="stylesheet" href="../Shared/styles.css">

    </head> 
        
    
    <body id="stage" class="coloured-bg">
        <div id="tuckman"></div> 
        <div id="container"></div>
        <div id="comfort"></div>

    </body>
    <script src="../3rdParty/require.min.js" data-main="js/compiled"></script>
</html>