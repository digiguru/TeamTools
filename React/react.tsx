/*
<ul id="users"><li><span class="user">Adam Hall</span><a href="void(0);">X</a></li><li><span class="user">Billie Davey</span><a href="void(0);">X</a></li><li><span class="user">Laura Rowe</span><a href="void(0);">X</a></li><li><span class="user">Simon Dawson</span><a href="void(0);">X</a></li><li><span class="user">Fred</span><a href="void(0);">X</a></li></ul>
*/
class Delete extends React.Component<any, any> {
    render() {
        return <a href="void(0);">X</a>;
    }
}
class User extends React.Component<any, any> {
  render() {
    return <li><span className="user">{this.props.username}</span><Delete /></li>;
  }
}
class UserList extends React.Component<any, any> {
    render() {
        let users = [];

        this.props.users.forEach(function(user) {
            users.push(<User user={user} username={user.name} key={user.name} />);
        });
        return <ul id="users">{users}</ul>;
    }
}
let USERS = [
    {name: "bob"},
    {name: "Donald"}
];
ReactDOM.render(
  <UserList users={USERS} />,
  document.getElementById("container")
);