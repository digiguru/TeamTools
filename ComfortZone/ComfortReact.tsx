import * as React from "react";

const 
    styleStandard: React.CSSProperties = {
        fill: "rgb(0, 215, 254);"
    },
    styleHighlight: React.CSSProperties = {
        fill: "rgb(0, 180, 219);"
    };

export interface iComfortState {
    style: React.CSSProperties;
}

export class ChaosArea extends React.Component<any, iComfortState> {
     constructor(props) {
        super(props);       
        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {style: styleStandard};
    }

    _onMouseEnter() {
        this.setState({style: styleHighlight});
    }

    _onMouseLeave() {
        this.setState({style: styleStandard});
    }

    render() {
        return <g>
            <rect className="area js-area-standard" id="chaos" width="800" height="800" style="{this.state.style}"></rect>
            <text className="area-label" id="label-choas" text-anchor="middle" x="400" y="50">chaos</text>
        </g>;
    }
}

export class StretchArea extends React.Component<any, iComfortState> {
    constructor(props) {
        super(props);       
        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {style: styleStandard};
    }

    _onMouseEnter() {
        this.setState({style: styleHighlight});
    }

    _onMouseLeave() {
        this.setState({style: styleStandard});
    }

    render() {
        return <g>
            <circle className="area js-area-standard" id="stretch" r="300.12599083141845" cx="400" cy="400"  style="{this.state.style}"></circle>
            <text className="area-label" id="label-stretch" text-anchor="middle" x="400" y="200">stretch</text>
        </g>;
    }
}

export class ComfortArea extends React.Component<any, iComfortState> {
     constructor(props) {
        super(props);

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {style: styleStandard};
    }

    _onMouseEnter() {
        this.setState({style: styleHighlight});
    }

    _onMouseLeave() {
        this.setState({style: styleStandard});
    }

    render() {
        return <g>
            <circle className="area js-area-standard" id="comfort" r="97.516770592797" cx="400" cy="400" style="{this.state.style}"></circle>
            <text className="area-label" id="label-comfort" text-anchor="middle" x="400" y="400">comfort</text>
        </g>;
    }
}
export class Stage extends React.Component<any, any> {
    render() {
        return <svg id="stage" width="800" height="800">{this.props.children}</svg>;
    }
    public static  full(props) {
        return <svg id="stage" width="800" height="800">{props.children}</svg>;
    }
}
export class ComfortReact extends React.Component<any, iComfortState> {

    constructor(props) {
        super(props);
        this.state = {style: styleStandard};
    }

    _onMouseEnter() {
        this.setState({style: styleHighlight});
    }

    _onMouseLeave() {
        this.setState({style: styleStandard});
    }

    render() {
        return <Stage>
                    <ChaosArea />
                    <g id="zones">
                        <StretchArea />
                        <ComfortArea />
                    </g>
                    <g id="history">
                    </g>
                    <rect id="clickable" width="800" height="800" fill-opacity="0.0"></rect>
                </Stage>;
    }

}