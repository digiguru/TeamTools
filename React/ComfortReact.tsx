import * as React from "react";

export interface IResizableModel {
    width: string;
    height: string;
}
export interface IInteractiveModelState {
    focus: string;
}
export interface IResizableInteractiveModelState extends IInteractiveModelState  {
    focus: string;
    width: string;
    height: string;
}

export class ChaosArea extends React.Component<any, IResizableInteractiveModelState> {
     constructor(props) {
        super(props);
        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);
        this.state = {
            focus: "not-in-focus",
            width: props.width || "100%",
            height: props.height || "100%"
        };
    }

    _onMouseEnter() {
        this.setState({focus: "in-focus"});
    }

    _onMouseLeave() {
        this.setState({focus: "not-in-focus"});
    }

    render() {
        const className: string = this.state.focus + " area js-area-standard";
        return <g>
            <rect className={className} onMouseEnter={this._onMouseEnter} onMouseLeave={this._onMouseLeave} id="chaos" width={this.state.width} height={this.state.height}></rect>
            <text className="area-label" id="label-choas" text-anchor="middle" textAnchor="middle" x="50%" y="20">chaos</text>
        </g>;
    }
}

export class StretchArea extends React.Component<any, IResizableInteractiveModelState> {
    constructor(props) {
        super(props);

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {
            focus: "not-in-focus",
            width: props.width || "100%",
            height: props.height || "100%"
        };
    }

    _onMouseEnter() {
        this.setState({focus: "in-focus"});
    }

    _onMouseLeave() {
        this.setState({focus: "not-in-focus"});
    }

    render() {
        const className: string = this.state.focus + " area js-area-standard";
        return <g>
            <circle 
                onMouseEnter={this._onMouseEnter} 
                onMouseLeave={this._onMouseLeave} 
                className={className} id="stretch" r="0%" cx="50%" cy="50%">
                {this.props.children}
            </circle>
            <text className="area-label" id="label-stretch" text-anchor="middle" textAnchor="middle" x="50%" y="20%">stretch</text>
        </g>;
    }
}
export class BouncyAnimation extends React.Component<any, any> {
     render() {
         const delay = this.props.delay || "0s";
         const duration = this.props.duration || "0.8s";
         const toValue = parseInt(this.props.value || (20), 10);
         const toValueType = this.props.valueType || "%";
         const values = [
            0 + toValueType,
            (toValue + toValue / 4) + toValueType,
            (toValue - toValue / 10) + toValueType,
            (toValue + toValue / 20) + toValueType,
            (toValue) + toValueType,
         ];
         const valuesToString = values.join(";");
         return <animate attributeType="XML" attributeName="r" from="0%" to="20%" 
                    dur={duration} 
                    begin={delay} 
                    values={valuesToString}
                    keyTimes="0; 0.3; 0.6; 0.8; 1"
                    fill="freeze"
                    />;
     }
}
export class ComfortArea extends React.Component<any, IResizableInteractiveModelState> {
     constructor(props) {
        super(props);

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {
            focus: "not-in-focus",
            width: props.width || "100%",
            height: props.height || "100%"
        };
    }

    _onMouseEnter() {
        this.setState({focus: "in-focus"});
    }

    _onMouseLeave() {
        this.setState({focus: "not-in-focus"});
    }

    render() {
        const className: string = this.state.focus + " area js-area-standard";
        //const startValue = 0 || this.props.value;
        const value = this.props.value || 20;
        return <g>
            <circle onMouseEnter={this._onMouseEnter}
                onMouseLeave={this._onMouseLeave}
                className={className} id="comfort" r="0%" cx="50%" cy="50%">
                {this.props.children}
            </circle>
            <text className="area-label" id="label-comfort" text-anchor="middle" textAnchor="middle" x="50%" y="50%">comfort</text>
        </g>;
        // keySplines=".42 0 1 1;0 0 .59 1;.42 0 1 1;0 0 .59 1;.42 0 1 1;0 0 .59 1;.42 0 1 1;0 0 .59 1;" 
    
    }
}
export class Stage extends React.Component<any, any> {

    constructor(props) {
        super(props);
        this.state = {
            width: "800",
            height: "800"
        };
    }

    render() {
        return <svg id="stage" width={this.state.width} height={this.state.height}>{this.props.children}</svg>;
    }
}
export class ComfortReact extends React.Component<any, any> {

    render() {
        return <Stage>
                    <ChaosArea />
                    <g id="zones">
                        <StretchArea><BouncyAnimation value="45" /></StretchArea>
                        <ComfortArea><BouncyAnimation value="20" delay="0.5s" /></ComfortArea>
                    </g>
                    <g id="history">
                    </g>
                </Stage>;
                //         <rect id="clickable" width="100%" height="100%" fill-opacity="0.0"></rect>
            
    }

}