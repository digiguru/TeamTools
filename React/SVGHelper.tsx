
import * as React from "react";
import {Point} from "Point";
/*require(["../Coords/Polar"], (u) => {
    console.log(new u.Polar(1,20));
});*/
//const r = new Polar(1,20);

export class SVGEvents {
    public static getDistnace(x: number, y: number, target: SVGElement) {
        return Point.distance(new Point(x, y), SVGEvents.getCenter(target));
    }
    public static getCenter(target: SVGElement) {
        const rect = target.getBoundingClientRect();
        return new Point(
            rect.left + (rect.width / 2),
            rect.top + (rect.height / 2)
        )
        /*
        return {
            x:rect.left + (rect.width / 2),
            y:rect.top + (rect.height / 2)
        }
        */
    }
}

export class Events {

    public static mouseEnter() {
        this.setState({focus: "in-focus"});
    }

    public static mouseDown() {
        this.setState({focus: "active"});
    }

    public static mouseUp(a) {
        console.log(SVGEvents.getCenter(a.target));
        this.setState({focus: "not-in-focus"});
    }

    public static mouseLeave() {
        this.setState({focus: "not-in-focus"});
    }

}

export class Stage extends React.Component<any, any> {

    constructor(props) {
        super(props);
        this.state = {
            width: "800",
            height: "800"
        };
    }

    render() {
        return <svg id="stage" width={this.state.width} height={this.state.height}>
            {this.props.children}
        </svg>;
    }
}

export class BouncyAnimation extends React.Component<any, any> {
     render() {
         const delay = this.props.delay || "0s";
         const duration = this.props.duration || "0.8s";
         const toValue = parseInt(this.props.value || (20), 10);
         const toValueType = this.props.valueType || "%";
         const values = [
            0 + toValueType,
            (toValue + toValue / 4) + toValueType,
            (toValue - toValue / 10) + toValueType,
            (toValue + toValue / 20) + toValueType,
            (toValue) + toValueType,
         ];
         const valuesToString = values.join(";");
         return <animate attributeType="XML" attributeName={this.props.attributeName} from="0%" to="20%"
                    dur={duration}
                    begin={delay}
                    values={valuesToString}
                    keyTimes="0; 0.3; 0.6; 0.8; 1"
                    fill="freeze"
                    />;
     }
}
