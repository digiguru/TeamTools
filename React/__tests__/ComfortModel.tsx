import * as React from "react";
import {ChaosArea, StretchArea, ComfortArea, ComfortReact} from "../ComfortReact";
import {Stage, BouncyAnimation} from "../SVGHelper";
const renderizer = require("react-test-renderer");

it("Should show the chaos area", () => {
    // Arrange
    const component = renderizer.create(
        <Stage><ChaosArea></ChaosArea></Stage>
    );
    const tree = component.toJSON();
    const mouseOverArea = tree.children[0].children[0];
    expect(tree).toMatchSnapshot();

    // Act snapshot 2
    mouseOverArea.props.onMouseEnter();
    // Assert Snapshot 2
    
    expect(component.toJSON()).toMatchSnapshot();

    // Act snapshot 3
    mouseOverArea.props.onMouseLeave();
    // Assert Snapshot 3
    expect(component.toJSON()).toMatchSnapshot();

});

it("Should show the stretch area", () => {
    const component = renderizer.create(
        <Stage><StretchArea></StretchArea></Stage>
    );
    const tree = component.toJSON();
    const mouseOverArea = tree.children[0].children[0];
    expect(tree).toMatchSnapshot();

    // Act snapshot 2
    mouseOverArea.props.onMouseEnter();
    // Assert Snapshot 2
    expect(component.toJSON()).toMatchSnapshot();

    // Act snapshot 3
    mouseOverArea.props.onMouseLeave();
    // Assert Snapshot 3
    expect(component.toJSON()).toMatchSnapshot();

});


it("Should show the comfort area - animated", () => {
    const component = renderizer.create(
        <Stage><ComfortArea><BouncyAnimation attributeName="r" /></ComfortArea></Stage>
    );
    expect(component.toJSON()).toMatchSnapshot();

});

it("Should show the comfort area - standard", () => {
    const component = renderizer.create(
        <Stage><ComfortArea></ComfortArea></Stage>
    );
    const tree = component.toJSON();
    
    const mouseOverArea = tree.children[0].children[0];
    expect(tree).toMatchSnapshot();

    // Act snapshot 2
    mouseOverArea.props.onMouseEnter();
    // Assert Snapshot 2
    expect(component.toJSON()).toMatchSnapshot();

    // Act snapshot 3
    mouseOverArea.props.onMouseLeave();
    // Assert Snapshot 3
    expect(component.toJSON()).toMatchSnapshot();

});
it("The complete comfort model", () => {
    const component = renderizer.create(
        <Stage width="800" height="800"><ComfortReact></ComfortReact></Stage>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});

