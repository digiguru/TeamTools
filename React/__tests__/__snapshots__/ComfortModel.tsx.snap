exports[`test Should show the chaos area 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <rect
      className="not-in-focus area js-area-standard"
      height="100%"
      id="chaos"
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      width="100%" />
    <text
      className="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20">
      chaos
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <rect
      class="not-in-focus area js-area-standard"
      height="100%"
      id="chaos"
      width="100%"
    >
    </rect>
    <text
      class="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20"
    >
      chaos
    </text>
  </g>
</svg>
`;

exports[`test Should show the chaos area 2`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <rect
      className="in-focus area js-area-standard"
      height="100%"
      id="chaos"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      width="100%" />
    <text
      className="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20">
      chaos
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <rect
      class="in-focus area js-area-standard"
      height="100%"
      id="chaos"
      width="100%"
    >
    </rect>
    <text
      class="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20"
    >
      chaos
    </text>
  </g>
</svg>
`;

exports[`test Should show the chaos area 3`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <rect
      className="not-in-focus area js-area-standard"
      height="100%"
      id="chaos"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      width="100%" />
    <text
      className="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20">
      chaos
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <rect
      class="not-in-focus area js-area-standard"
      height="100%"
      id="chaos"
      width="100%"
    >
    </rect>
    <text
      class="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20"
    >
      chaos
    </text>
  </g>
</svg>
`;

exports[`test Should show the comfort area - animated 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <circle
      className="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="comfort"
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      r="0%">
      <animate
        attributeName="r"
        attributeType="XML"
        begin="0s"
        dur="0.8s"
        fill="freeze"
        from="0%"
        keyTimes="0; 0.3; 0.6; 0.8; 1"
        to="20%"
        values="0%;25%;18%;21%;20%" />
    </circle>
    <text
      className="area-label"
      id="label-comfort"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%">
      comfort
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <circle
      class="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="comfort"
      r="0%"
    >
      <animate
        attributeName="r"
        attributeType="XML"
        begin="0s"
        dur="0.8s"
        fill="freeze"
        from="0%"
        keyTimes="0; 0.3; 0.6; 0.8; 1"
        to="20%"
        values="0%;25%;18%;21%;20%"
      >
      </animate>
    </circle>
    <text
      class="area-label"
      id="label-comfort"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%"
    >
      comfort
    </text>
  </g>
</svg>
`;

exports[`test Should show the comfort area - standard 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <circle
      className="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="comfort"
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      r="0%" />
    <text
      className="area-label"
      id="label-comfort"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%">
      comfort
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <circle
      class="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="comfort"
      r="0%"
    >
    </circle>
    <text
      class="area-label"
      id="label-comfort"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%"
    >
      comfort
    </text>
  </g>
</svg>
`;

exports[`test Should show the comfort area - standard 2`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <circle
      className="in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="comfort"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      r="0%" />
    <text
      className="area-label"
      id="label-comfort"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%">
      comfort
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <circle
      class="in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="comfort"
      r="0%"
    >
    </circle>
    <text
      class="area-label"
      id="label-comfort"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%"
    >
      comfort
    </text>
  </g>
</svg>
`;

exports[`test Should show the comfort area - standard 3`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <circle
      className="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="comfort"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      r="0%" />
    <text
      className="area-label"
      id="label-comfort"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%">
      comfort
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <circle
      class="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="comfort"
      r="0%"
    >
    </circle>
    <text
      class="area-label"
      id="label-comfort"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%"
    >
      comfort
    </text>
  </g>
</svg>
`;

exports[`test Should show the stretch area 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <circle
      className="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="stretch"
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      r="0%" />
    <text
      className="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%">
      stretch
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <circle
      class="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="stretch"
      r="0%"
    >
    </circle>
    <text
      class="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%"
    >
      stretch
    </text>
  </g>
</svg>
`;

exports[`test Should show the stretch area 2`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <circle
      className="in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="stretch"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      r="0%" />
    <text
      className="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%">
      stretch
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <circle
      class="in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="stretch"
      r="0%"
    >
    </circle>
    <text
      class="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%"
    >
      stretch
    </text>
  </g>
</svg>
`;

exports[`test Should show the stretch area 3`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <circle
      className="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="stretch"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      r="0%" />
    <text
      className="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%">
      stretch
    </text>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <circle
      class="not-in-focus area js-area-standard"
      cx="50%"
      cy="50%"
      id="stretch"
      r="0%"
    >
    </circle>
    <text
      class="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%"
    >
      stretch
    </text>
  </g>
</svg>
`;

exports[`test The complete comfort model 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <svg
    height="800"
    id="stage"
    width="800">
    <g>
      <rect
        className="not-in-focus area js-area-standard"
        height="100%"
        id="chaos"
        onMouseDown={undefined}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        onMouseUp={undefined}
        width="100%" />
      <text
        className="area-label"
        id="label-chaos"
        text-anchor="middle"
        textAnchor="middle"
        x="50%"
        y="20">
        chaos
      </text>
    </g>
    <g
      id="zones">
      <g>
        <circle
          className="not-in-focus area js-area-standard"
          cx="50%"
          cy="50%"
          id="stretch"
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onMouseUp={undefined}
          r="0%">
          <animate
            attributeName="r"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;56.25%;40.5%;47.25%;45%" />
        </circle>
        <text
          className="area-label"
          id="label-stretch"
          text-anchor="middle"
          textAnchor="middle"
          x="50%"
          y="20%">
          stretch
        </text>
      </g>
      <g>
        <circle
          className="not-in-focus area js-area-standard"
          cx="50%"
          cy="50%"
          id="comfort"
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onMouseUp={undefined}
          r="0%">
          <animate
            attributeName="r"
            attributeType="XML"
            begin="0.5s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;25%;18%;21%;20%" />
        </circle>
        <text
          className="area-label"
          id="label-comfort"
          text-anchor="middle"
          textAnchor="middle"
          x="50%"
          y="50%">
          comfort
        </text>
      </g>
    </g>
    <g
      display={undefined}
      id="history" />
  </svg>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <svg
    height="800"
    id="stage"
    width="800"
  >
    <g>
      <rect
        class="not-in-focus area js-area-standard"
        height="100%"
        id="chaos"
        width="100%"
      >
      </rect>
      <text
        class="area-label"
        id="label-chaos"
        text-anchor="middle"
        textAnchor="middle"
        x="50%"
        y="20"
      >
        chaos
      </text>
    </g>
    <g id="zones">
      <g>
        <circle
          class="not-in-focus area js-area-standard"
          cx="50%"
          cy="50%"
          id="stretch"
          r="0%"
        >
          <animate
            attributeName="r"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;56.25%;40.5%;47.25%;45%"
          >
          </animate>
        </circle>
        <text
          class="area-label"
          id="label-stretch"
          text-anchor="middle"
          textAnchor="middle"
          x="50%"
          y="20%"
        >
          stretch
        </text>
      </g>
      <g>
        <circle
          class="not-in-focus area js-area-standard"
          cx="50%"
          cy="50%"
          id="comfort"
          r="0%"
        >
          <animate
            attributeName="r"
            attributeType="XML"
            begin="0.5s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;25%;18%;21%;20%"
          >
          </animate>
        </circle>
        <text
          class="area-label"
          id="label-comfort"
          text-anchor="middle"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          comfort
        </text>
      </g>
    </g>
    <g id="history"></g>
  </svg>
</svg>
`;
